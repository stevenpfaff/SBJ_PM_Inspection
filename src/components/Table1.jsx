import { Table } from 'react-bootstrap'

import data from '../data/table1.json'

const NUMBER_FORMAT = Intl.NumberFormat('en-US', {
  style: 'unit',
  unit: 'mile',
  unitDisplay: 'short'
})

const formatNumber = value => NUMBER_FORMAT.format(value)

const MINS_IN_HOUR = 60

const formatHours = hours => {
  if (Number.isNaN(hours)) return ''

  const minutes = hours * MINS_IN_HOUR

  const wholeHours = Math.floor(minutes / MINS_IN_HOUR)

  if (wholeHours) {
    if (wholeHours === 1) {
      return `${wholeHours} hr`
    }

    return `${wholeHours} hrs`
  }

  return ''
}

const formatMinutes = hours => {
  if (Number.isNaN(hours)) return ''

  const minutes = hours * MINS_IN_HOUR

  const wholeMinutes = Math.floor(minutes % MINS_IN_HOUR)

  if (wholeMinutes) {
    if (wholeMinutes === 1) {
      return `${wholeMinutes} min`
    }

    return `${wholeMinutes} mins`
  }

  return ''
}

const Table1 = () => (
  <Table striped bordered hover>
    <thead>
      <tr>
        <th>Customer</th>
        <th>Unit #</th>
        <th>VIN</th>
        <th>Distance</th>
        <th>Duration</th>
      </tr>
    </thead>
    <tbody>
      {data.map((row, idx) => (
        <tr key={idx}>
          <td>{row.customer}</td>
          <td className='fixed-width'>{row.unitNumber}</td>
          <td className='fixed-width'>{row.vin}</td>
          <td className='distance'>{formatNumber(row.miles)}</td>
          <td className='duration'>
            <span>{formatHours(row.hours)}</span>
            <span>{formatMinutes(row.hours)}</span>
          </td>
        </tr>
      ))}
    </tbody>
  </Table>
)

export default Table1
