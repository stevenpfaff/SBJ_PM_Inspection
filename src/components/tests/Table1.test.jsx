import { render } from '@testing-library/react'

import Table1 from '../Table1'

describe('Table1', () => {
  it('should match snapshot', () => {
    const { container } = render(<Table1 />)

    expect(container).toMatchSnapshot()
  })
})
