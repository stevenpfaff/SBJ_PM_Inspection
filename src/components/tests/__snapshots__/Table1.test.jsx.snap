// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table1 should match snapshot 1`] = `
<div>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Customer
        </th>
        <th>
          Unit #
        </th>
        <th>
          VIN
        </th>
        <th>
          Distance
        </th>
        <th>
          Duration
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jared Walters
        </td>
        <td
          class="fixed-width"
        >
          RP4ZJ0WPXX
        </td>
        <td
          class="fixed-width"
        >
          1G4GJ11Y9HP422546
        </td>
        <td
          class="distance"
        >
          300 mi
        </td>
        <td
          class="duration"
        >
          <span>
            25 hrs
          </span>
          <span>
            13 mins
          </span>
        </td>
      </tr>
      <tr>
        <td>
          Stewart Garnett
        </td>
        <td
          class="fixed-width"
        >
          EFUK4D1K4M
        </td>
        <td
          class="fixed-width"
        >
          2GCEC19Z1S1244154
        </td>
        <td
          class="distance"
        >
          45 mi
        </td>
        <td
          class="duration"
        >
          <span>
            25 hrs
          </span>
          <span>
            38 mins
          </span>
        </td>
      </tr>
      <tr>
        <td>
          Bruce Harvey
        </td>
        <td
          class="fixed-width"
        >
          YGM7GUJ7BJ
        </td>
        <td
          class="fixed-width"
        >
          WMWZN3C51BT133317
        </td>
        <td
          class="distance"
        >
          125 mi
        </td>
        <td
          class="duration"
        >
          <span>
            25 hrs
          </span>
          <span>
            20 mins
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
